pkg/permissions/set.go

```diff
@@ -137,3 +138,25 @@ func (ps *Set) IsSubSetOf(parent Set) bool {
 
 	return true
 }
+
+// HasSameRules returns true if the two sets have exactly the same rules.
+func (ps Set) HasSameRules(other Set) bool {
```

Comment:
The aim is to check that they are identical, i.e. exactly the same rules. I'll add a check to see if there is the same number of rules in both, which should do what I want.